Phylogenetic data prep training (#3497)
galaxyproject/training-material: topics/ecology/tutorials/phylogeny-data-prep/tutorial.md

key_points:
- You now are able to
- Predict proteins in a nucleotide sequence *de-novo* using **funannotate_predict**
- Find orthologs across different samples with **orthofinder**
- Align orthologs with **ClustalW** in preparation for phylogeny reconstruction <!-- link to phylogeny reconstruction training. -->
contributors:
- roncoronimiguel
- brigidagallone

---

A robust and well-resolved phylogenetic classification is essential to understand genetic relationships within and between species and the evolution of their phenotypic diversity. In the last decade the genomic revolution has represented a drastic change in the amount of data used for phylogenetic inference. The single-gene approach using universal phylogenetic markers for the different lineages across the tree of life, is now being replaced by the assembly of taxon-rich and genome-scale data matrices, the so called phylogenomic approach.

Molecular sequence data can be used to construct a phylogeny by comparing differences between nucleotide or amino acid sequences across species or strains, a technique called phylogenomics. {% cite Young2019 %} have written a comprehensive review on the topic of phylogenomics.

> In this tutorial we prepare genetic sequence data for phylogenetic reconstruction.
> We will use sequences from chromosome 5 of five strains of the yeast *Saccharomyces cerevisiae*. This requires first the prediction of protein coding genes from the genome. We use Funannotate to predict protein-coding genes . Next, we find the proteins that are present in more than one genome, called orthologs, using Proteinortho, and extract a subset with orthologs that are present in all samples. Finally, each set of orthologs is aligned using ClustalW. The resulting dataset is ready to be used for phylogenetic reconstruction.

Reviewer: ```suggestion

In this tutorial we prepare genetic sequence data for phylogenetic reconstruction, using sequences from chromosome 5 of five strains of the yeast *Saccharomyces cerevisiae*. 

To prepare the data, we will:
1. Predict protein coding genes from the genome using Funannotate
2. Find the proteins that are present in more than one genome, called orthologs, using Proteinortho and extract a subset with orthologs that are present in all samples. 
4. Align each set of orthologs using ClustalW.

The resulting dataset is ready to be used for phylogenetic reconstruction.
```

> **If you are starting from sequence reads, please follow

Reviewer: Could you put that as requirements in the tutorial metadata?

>     -> `{{ phylogenetic-data-prep }}`):
>
>    ```
>    https://zenodo.org/record/6610704/files/BK006939.2.genbank
>    https://zenodo.org/record/6610704/files/BK006939.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM000925.1.genbank
>    https://zenodo.org/record/6610704/files/CM000925.1.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005043.2.genbank
>    https://zenodo.org/record/6610704/files/CM005043.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005299.1.genbank
>    https://zenodo.org/record/6610704/files/CM005299.1.genome.fasta
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>
> 3. Optional: Rename each dataset to its accession number followed by '.genome.fasta' or '.genbank'.
> 4. Group the datasets into [collections](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html). These will ease data handling and help minimize the clutter in your history. Make a collection of nucleotide sequences and another of protein sequences.
>
>    {% snippet faqs/galaxy/collections_build_list.md %}
>
>
{: .hands_on}

> # Genome annotation

Reviewer: ```suggestion
# Protein coding gene prediction
```

Reviewer: To stay consistent with the introduction

## Mask repetitive sequences

> Before we can annotate the genome, we will prepare the data by masking repetitive sequences in the genome.

Reviewer: ```suggestion

Before we can annotate the genome, we will prepare the data by masking repetitive sequences in the genome.
```

We use **RepeatMasker**
{% cite RepeatMasker %}, a program that screens DNA sequences for interspersed repeats and low complexity DNA sequences.
This program has historically made use of [RepBase](https://www.girinst.org/repbase/update/index.html) ({%cite Kohany2006-ks %}), a service of the Genetic Information Research Institute, but this database in no longer open access. Instead, we will use [Dfam](https://www.dfam.org/home) ({%cite Storer2021 %}) an open collection of Transposable Element DNA sequence alignments,  HMMs derived from Repbase sequences and consensus sequences. For this reason, the annotation of repetitive sequences might be incomplete.

RepeatMasker will only accept compact fasta headers. Before we can mask repetitive regions with RepeatMasker we must trim the NCBI long header (`BK006939.2 TPA_inf: Saccharomyces cerevisiae S288C chromosome V, complete sequence`) to leave only the accession number (`>BK006939.2`) by using a regular expression.

> > ### {% icon hands_on %} Hands-on: Mask repetitive sequences

Reviewer: ```suggestion

> ### {% icon hands_on %} Hands-on: Mask repetitive sequences
```

>
> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2) %} with the following parameters:
>    - {% icon param-collection %} *"File to process"*: `output` (Input dataset collection)
>    - In *"Replacement"*:
>        - {% icon param-repeat %} *"Insert Replacement"*
>            - *"Find pattern"*: `(>[^ ]+).+`

> >            - *"Replace with:"*: `\1`
> >
> >    > ### {% icon comment %} Comment
> >    >
> >    >`\1` replaces the text in the header with the first matched text in the header, the accession number in this case.
> >    {: .comment}

Reviewer: ```suggestion
>            - *"Replace with:"*: `\1`
>
>               > ### {% icon comment %} Comment
>               >
>               >`\1` replaces the text in the header with the first matched text in the header, the accession number in this case.
>               {: .comment}
```

>
> 2. {% tool [RepeatMasker](toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/4.1.2-p1+galaxy0) %} with the following parameters:
>    - {% icon param-file %} *"Genomic DNA"*: `outfile` (output of **Replace Text** {% icon tool %})
>    - *"Repeat library source"*: `DFam (curated only, bundled with RepeatMasker)`
>        - *"Select species name from a list?"*: `No`
>            - *"Repeat source species"*: `"Saccharomyces cerevisiae"`
>
>    > ### {% icon comment %} Comment
>    >

> >    > In principal, all unique clade names occurring in [NCBI taxonomy database](https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html) can be used for species. Capitalization is ignored, multiple words need to bound by apostrophes. Not all "common" English names occur in the taxonomy database. Using Latin names is always safest.

Reviewer: What do you mean here? Is it a comment related to RepeatMasker or how to prepare data for RepeatMasker?

>    {: .comment}
>
{: .hands_on}

> 

Reviewer: Is there anything to inspect or check after? It would be useful for learners

Reviewer: Also some comments on the outputs to make a transition with the next step

## Annotate with Funannotate

> We will predict protein-coding genes from genomic sequences using [Funannotate](https://funannotate.readthedocs.io/) ({% cite Young2019 %}), which collects evidence from different ab-initio gene predictors as well as from RNA-seq or ESTs data. Funannotate has been developed for Fungi but it works with any Eukaryotic genome. The output of Funannotate is a list of ORFs and their translation in GenBank format.

Reviewer: ```suggestion

We will predict protein-coding genes from genomic sequences using [Funannotate](https://funannotate.readthedocs.io/) ({% cite Young2019 %}), which collects evidence from different ab-initio gene predictors as well as from RNA-seq or ESTs data. Funannotate has been developed for Fungi but it works with any Eukaryotic genome. The output of Funannotate is a list of ORFs and their translation in GenBank format.
```

> If you would like to learn about genome annotation in more depth, the GTN has a [section]({{ site.baseurl }}/topics/genome-annotation) dedicated to training on genome annotation, including a hands-on tutorial on [Funannotate]({% link topics/genome-annotation/tutorials/funannotate/tutorial.md %}).

Reviewer: I would put that as a comment

{: .warning}

> ### {% icon hands_on %} Hands-on: Annotate genome
>
> 1. {% tool [Funannotate predict annotation](toolshed.g2.bx.psu.edu/repos/iuc/funannotate_predict/funannotate_predict/1.8.9+galaxy2) %} with the following parameters:
>    - {% icon param-file %} *"Assembly to annotate"*: `output_masked_genome` (output of **RepeatMasker** {% icon tool %})
>    - In *"Organism"*:
>        - *"Name of the species to annotate"*: `Saccharomyces cerevisiae`
>        - *"Is it a fungus species?"*: `Yes`
>        - *"Ploidy of assembly"*: `1`
>    - In *"Evidences"*:
>        - *"Select protein evidences"*: `Use UniProtKb/SwissProt (from selected Funannotate database)`
>    - In *"Busco"*:
>        - *"BUSCO models to align"*: `saccharomycetes`
>        - *"Initial Augustus species training set for BUSCO alignment"*: `saccharomyces`
>    - In *"Augustus settings (advanced)"*:
>        - *"Minimum number of models to train Augustus"*: `15`
>    
>
>    > ### {% icon tip %} Tip
>    >

> >    > If available, include mRNA and/or ESTs evidence to increase sensitivity of predictions.

Reviewer: Which parameters should be selected for that?

>    {: .tip}
>
>    > ### {% icon comment %} Comment
>    >

> >    > When annotating full genomes, increase the number of *'Minimum number of models to train Augustus'* to an appropriate value. For the small sample dataset used here, values larger than 15 will result in failure.

Reviewer: I would move that just below the "Minimum number of models to train Augustus" parameter

>    {: .comment}
{: .hands_on}

> 

Reviewer: Is there anything to inspect or check after? It would be useful for learners

Reviewer: Also some comments on the outputs to make a transition with the next step

Author: added file inspection and transition

## Extract ORFs into FASTA files

> In this step, we extract the protein sequences from the GenBank files into multi-FASTA files. Additionally, we modify the headers to include the accession number of the sample. This creates an unique accession-proteinID header for each predicted protein, and will allow us to retrieve them after we combine all predicted proteins into one multi-FASTA file.

Reviewer: ```suggestion

In this step, we extract the protein sequences from the GenBank files into multi-FASTA files. Additionally, we modify the headers to include the accession number of the sample. This creates an unique accession-proteinID header for each predicted protein, and will allow us to retrieve them after we combine all predicted proteins into one multi-FASTA file.
```

>    - {% icon param-file %} *"gene bank file"*: `annot_gbk` (output of **Funannotate predict annotation** {% icon tool %} - or the genbank files downloaded from Zenodo)
>
> 2. {% tool [Regex Find And Replace](toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.1) %} with the following parameters:
>    - {% icon param-file %} *"Select lines from"*: `aa_output` (output of **Extract ORF** {% icon tool %})
>    - In *"Check"*:
>        - {% icon param-repeat %} *"Insert Check"*
>            - *"Find Regex"*: `>([^ ]+).+`
>            - *"Replacement"*: `>#{input_name}_\1`
> 
> 1. {% tool [Collapse Collection](toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2) %} with the following parameters:
>    - {% icon param-file %} *"Collection of files to collapse into single dataset"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Prepend File name"*: `Yes`
>
{: .hands_on}

> ### {% icon question %} Questions
>
> 1. Is the number of predicted ORFs the same across the samples?
>
> > ### {% icon solution %} Solution
> >
> > 1. No, the number of predicted ORFs ranges from 193 to 199.
> >
> {: .solution}
>
{: .question}

> 
> 
> 
> 
> 

Reviewer: ```suggestion
```

Orthologs are genes in different species evolved from a common ancestral gene by a speciation (lineage-splitting) event and contain the information needed for building phylogenies. The input for this step is the collection of multi-FASTA extracted from the GenBank file and modified to have unique headers.
The result file 'orthology-groups' contains one row per orthogroup and one column per sample.

Next, we select orthogroups where all the species are represented by only one protein, 1:1 single copy orthologs (a total of 4 proteins per orthogroup for this dataset).

> ### {% icon hands_on %} Hands-on: find and filter orthologs
>
> 1. {% tool [Proteinortho](toolshed.g2.bx.psu.edu/repos/iuc/proteinortho/proteinortho/6.0.14+galaxy2.9.1) %} with the following parameters:
>    - {% icon param-file %} *"Select the input fasta files (>2)"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Activate synteny feature (POFF)"*: `no`
>
> 2. {% tool [Filter](Filter1) %} with the following parameters:
>    - {% icon param-file %} *"Filter"*: `proteinortho` (output of **Proteinortho** {% icon tool %})
>    - *"With following condition"*: `c1==4 and c2==4`
>

> {: .hands_on}

Reviewer: ```suggestion
> 3. Inspect the 'orthology-groups' tabular file.
{: .hands_on}
```

> > ### {% icon question %} Questions
> >
> >Look at the 'orthology-groups' tabular file.
> >
> > 1. Do any samples contain more than one gene for any given orthogroup?
> > 2. What is the name given to these homologous genes within the same genome?
> >
> > > ### {% icon solution %} Solution
> > >
> > > 1. Yes, *CM000925.1* contains two genes on the first orthogroup.
> > > 2. Paralogs. These are gene copies created by a duplication event within the same genome.
> > >
> > {: .solution}
> >
> {: .question}
> 
> 
> > ### {% icon question %} Questions
> > 
> >Look at the filtered orthogroups.
> > 1. How many orthogroups are represented once only in all four samples?
> >
> > > ### {% icon solution %} Solution
> > >
> > > 1. 173
> > >
> > {: .solution}
> >
> {: .question}

Reviewer: ```suggestion
> ### {% icon question %} Questions
>
>1. Inspect the the 'orthology-groups' tabular file.
>
>    1. Do any samples contain more than one gene for any given orthogroup?
>    2. What is the name given to these homologous genes within the same genome?
>
>    > ### {% icon solution %} Solution
>    >
>    > 1. Yes, *CM000925.1* contains two genes on the first orthogroup.
>    > 2. Paralogs. These are gene copies created by a duplication event within the same genome.
>    >
>    {: .solution}
>
> 2. Look at the filtered orthogroups.
>     1. How many orthogroups are represented once only in all four samples?
>
>     > ### {% icon solution %} Solution
>     >
>     > 1. 173
>     >
>     {: .solution}
>
{: .question}
```

## Quality control with **Busco**

> Busco is a dataset of nearl-universal, single-copy orthologs.

Reviewer: ```suggestion

Busco is a dataset of nearl-universal, single-copy orthologs.
```

Reviewer: BUSCO is a dataset, not a tool?

> Here we use it for QC of the assembly and annotation produced above.

Reviewer: ```suggestion
Here we use it for quality control of the assembly and annotation produced above.
```

It outputs a proxy of completeness, duplication and fragmentation of the annotation (Busco can also be used to assess the completeness of a genome assembly).

> 
> 
> 

Reviewer: ```suggestion
```

> ### {% icon hands_on %} QC with Busco
>
> 1. {% tool [Busco](toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/4.1.4) %} with the following parameters:
>    - {% icon param-file %} *"Sequences to analyse"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Mode"*: `Proteome`
>    - *"Lineage"*: `Saccharomycetes`
>    - In *"Advanced Options"*:
>        - *"Augustus species model"*: `Use the default species for selected lineage`
>
>    > ### {% icon tip %} Tip
>    > Make sure you select the 'lineage' closest to the species(s) you are analyzing.
>    >
>    {: .tip}
>
{: .hands_on}

> ### {% icon question %} Questions
>
> 1. The [Complete ('C') metric](https://busco.ezlab.org/busco_userguide.html#complete) stands for complete Busco genes identified. Look at the Busco output file 'Short summary' for sample BK006939.2. What is the 'C' number?
> 2. Why is it so low?
>
> > ### {% icon solution %} Solution
> >
> > 1. 4.0%
> > 2. Our dataset contains only chromosome 5 of the yeast genome.
> >
> {: .solution}
>
{: .question}

> 
> 
> 
> 

Reviewer: ```suggestion
```

## Extract proteins with Proteinortho

> Next we extract 1:1 single copy orthologs and generate one multi fasta file per ortholog.  

Reviewer: ```suggestion

Next we extract 1:1 single copy orthologs and generate one multi fasta file per ortholog.  
```

> The output is a collection of multi-fasta ortholog files. All species are represented in each file and are ready to be aligned.

Reviewer: I would move that after the hands-on part

> ### {% icon hands_on %} Hands-on: extract protein sequences
>
> 1. {% tool [Proteinortho grab proteins](toolshed.g2.bx.psu.edu/repos/iuc/proteinortho_grab_proteins/proteinortho_grab_proteins/6.0.14+galaxy2.9.1) %} with the following parameters:
>    - {% icon param-file %} *"Select the input fasta files"*: `output` (output of **Collapse Collection** {% icon tool %})
>    - *"Query type"*: `orthology-groups output file`
>        - {% icon param-file %} *"A orthology-groups file"*: `out_file1` (output of **Filter** {% icon tool %})
>
>
{: .hands_on}

> 
> 
> 

Reviewer: ```suggestion
```

# Align ortholog sequences

> Alignment of sequences allows cross-species (or other taxonomic level, strain, taxa) comparison. An alignment is a hypothesis of positional homology between bases/amino acids of different sequences. A correct alignment is crucial for phylogenetic inference. Here we use ClustaW, a well-established pairwise sequence aligner.

Reviewer: ```suggestion

Alignment of sequences allows cross-species (or other taxonomic level, strain, taxa) comparison. An alignment is a hypothesis of positional homology between bases/amino acids of different sequences. A correct alignment is crucial for phylogenetic inference. Here we use ClustaW, a well-established pairwise sequence aligner.
```

First we modify the headers of the multi-fasta file, such that only the sample name is retained. This is important for future conatenation of alignment into a supermatrix (see Maximum likelihood GTN training)

> ### {% icon hands_on %} Hands-on: Align orthologs with ClustalW
>
> 1. {% tool [Regex Find And Replace](toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.1) %} with the following parameters:
>    - {% icon param-file %} *"Select lines from"*: `listproteinorthograbproteins` (output of **Proteinortho grab proteins** {% icon tool %})
>    - In *"Check"*:
>        - {% icon param-repeat %} *"Insert Check"*
>            - *"Find Regex"*: `(>[^_]+).+`
>            - *"Replacement"*: `\1`
>
> 2. {% tool [ClustalW](toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1) %} with the following parameters:
>    - {% icon param-file %} *"FASTA file"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Data type"*: `Protein sequences`
>    - *"Output alignment format"*: `FASTA format`
>    - *"Output complete alignment (or specify part to output)"*: `Complete alignment`
>
{: .hands_on}

> 

Reviewer: It would be nice to comment the output, maybe inspect it or visualize it

# Conclusion

> 

Reviewer: Could you add a small conclusion?

Molecular sequence data can be used to construct a phylogeny by comparing differences between nucleotide or amino acid sequences across species or strains, a technique called phylogenomics. {% cite Young2019 %} have written a comprehensive review on the topic of phylogenomics.
In this tutorial we prepare genetic sequence data for phylogenetic reconstruction.
We will use sequences from chromosome 5 of five strains of the yeast *Saccharomyces cerevisiae*. This requires first the prediction of protein coding genes from the genome. We use Funannotate to predict protein-coding genes . Next, we find the proteins that are present in more than one genome, called orthologs, using Proteinortho, and extract a subset with orthologs that are present in all samples. Finally, each set of orthologs is aligned using ClustalW. The resulting dataset is ready to be used for phylogenetic reconstruction.

**If you are starting from sequence reads, please follow

> [An Introduction to Genome Assembly]({{ site.baseurl }}/topics/assembly/tutorials/general-introduction/tutorial.html), and the appropriate genome assembly training for your sequencing technology from GTN's [Assembly]({{ site.baseurl }}/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html) section**

Reviewer 2: ```suggestion
[An Introduction to Genome Assembly]({% link topics/assembly/tutorials/general-introduction/tutorial.md %}), and the appropriate genome assembly training for your sequencing technology from GTN's [Assembly]({% link topics/assembly/index.md %}) section**
```

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Create a new history for this tutorial
> 2. Import the files from [Zenodo]({{ https://zenodo.org/record/6610704#.Ypn3FzlBw5k }}) or from
>    the shared data library (`GTN - Material` -> `{{ ecology }}`
>     -> `{{ phylogenetic-data-prep }}`):
>
>    ```
>    https://zenodo.org/record/6610704/files/BK006939.2.genbank
>    https://zenodo.org/record/6610704/files/BK006939.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM000925.1.genbank
>    https://zenodo.org/record/6610704/files/CM000925.1.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005043.2.genbank
>    https://zenodo.org/record/6610704/files/CM005043.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005299.1.genbank
>    https://zenodo.org/record/6610704/files/CM005299.1.genome.fasta
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>
> 3. Optional: Rename each dataset to its accession number followed by '.genome.fasta' or '.genbank'.

> > 4. Group the datasets into [collections](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html). These will ease data handling and help minimize the clutter in your history. Make a collection of nucleotide sequences and another of protein sequences.

Reviewer 2: This upload step could be maybe simplified with the rule-based uploader.

You could provide a table like

```
https://zenodo.org/record/6610704/files/BK006939.2.genbank genbank 
https://zenodo.org/record/6610704/files/BK006939.2.genome.fasta fasta
https://zenodo.org/record/6610704/files/CM000925.1.genbank genbank
https://zenodo.org/record/6610704/files/CM000925.1.genome.fasta fasta
https://zenodo.org/record/6610704/files/CM005043.2.genbank genbank
https://zenodo.org/record/6610704/files/CM005043.2.genome.fasta fasta
https://zenodo.org/record/6610704/files/CM005299.1.genbank genbank
https://zenodo.org/record/6610704/files/CM005299.1.genome.fasta fasta
```

and then have students use the RBU to:

- build a dataset collection from table
- assign column 1 as url
- column 2  as filetype + collection name
- create a new column stripping the base path (it's an option in there somewhere)
- and use that new column 3 as the element identifier.

It's a small tradeoff of learning a little bit of complexity, but maybe it gives students more tools that can scale better for large datasets than manually renaming files / manually creating collections?

Author: I definitely see the learning value of doing it this way. But I think, in the end, it makes it harder for the student to get to the analysis part and is a bit beyond the point of this tutorial.

Reviewer 2: Ok, fine for me.

First we modify the headers of the multi-fasta file, such that only the sample name is retained. This is important for future conatenation of alignment into a supermatrix (see Maximum likelihood GTN training)

> ### {% icon hands_on %} Hands-on: Align orthologs with ClustalW
>
> 1. {% tool [Regex Find And Replace](toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.1) %} with the following parameters:
>    - {% icon param-file %} *"Select lines from"*: `listproteinorthograbproteins` (output of **Proteinortho grab proteins** {% icon tool %})
>    - In *"Check"*:
>        - {% icon param-repeat %} *"Insert Check"*
>            - *"Find Regex"*: `(>[^_]+).+`
>            - *"Replacement"*: `\1`
>
> 2. {% tool [ClustalW](toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1) %} with the following parameters:
>    - {% icon param-file %} *"FASTA file"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Data type"*: `Protein sequences`
>    - *"Output alignment format"*: `FASTA format`
>    - *"Output complete alignment (or specify part to output)"*: `Complete alignment`
>
{: .hands_on}

> ## Re-arrange
> 
> To create the template, each step of the workflow had its own subsection.
> 
> ***TODO***: *Re-arrange the generated subsections into sections or other subsections.
> Consider merging some hands-on boxes to have a meaningful flow of the analyses*

Reviewer 2: ```suggestion
```

I think this is done!

---
layout: tutorial_hands_on

title: Preparing genomic data for phylogeny reconstruction

> zenodo_link: https://zenodo.org/record/6610704#.Ypn3FzlBw5k

Reviewer 2: ```suggestion
zenodo_link: https://zenodo.org/record/6610704
```

questions:
- How do I find a set of common proteins (orthologs) across related species or strains?
- How do I organize a set of orthologs to infer evolutionary relations between species or strains (phylogenetic reconstruction)?
objectives:
- Mask repetitive elements from a genome
- Annotate (predict protein-coding genes) the genomes of the samples to compare
- Find a set of common proteins across the samples (orthologs)
- Align orthologs across samples
time_estimation: 3H
key_points:
- You now are able to
- Predict proteins in a nucleotide sequence *de-novo* using **funannotate_predict**
- Find orthologs across different samples with **orthofinder**
- Align orthologs with **ClustalW** in preparation for phylogeny reconstruction <!-- link to phylogeny reconstruction training. -->
contributors:
- roncoronimiguel

> - brigidagallone

Reviewer 2: If you like can use our fancy new contributions syntax to more accurately record contributions to a training material, declaring which people should be recorded for authorship or just editing. All of the sub-fields are optional, so if you just have authors/editors you can do just those two :) 

```
contributions:
  authorship: 
    - shiltemann
  editing:
    - bebatut
    - hexylena
  funding:
    - carpentries
  testing:
    - userX
  ux:
    - userY
  infrastructure:
    - userZ
```

see the [full gtn tutorial schema documentation](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/schemas/tutorial.html) for details on the individual roles.

---

> <!-- Add the references to the tutorials cited -->
> 
> # Introduction
> {:.no_toc}
> 
> <!-- This is a comment. -->

Reviewer 2: ```suggestion
```

>    https://zenodo.org/record/6610704/files/BK006939.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM000925.1.genbank
>    https://zenodo.org/record/6610704/files/CM000925.1.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005043.2.genbank
>    https://zenodo.org/record/6610704/files/CM005043.2.genome.fasta
>    https://zenodo.org/record/6610704/files/CM005299.1.genbank
>    https://zenodo.org/record/6610704/files/CM005299.1.genome.fasta
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>
> 3. Optional: Rename each dataset to its accession number followed by '.genome.fasta' or '.genbank'.
> 4. Group the datasets into [collections](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html). These will ease data handling and help minimize the clutter in your history. Make a collection of nucleotide sequences and another of protein sequences.
>
>    {% snippet faqs/galaxy/collections_build_list.md %}
>
>
{: .hands_on}

# Genome annotation

## Mask repetitive sequences
Before we can annotate the genome, we will prepare the data by masking repetitive sequences in the genome.

> Repeat-rich regions can interfere with genome annotation tools. In this step we find and soft-mask repetitive regions in the genome. The annotation tool can then take this information into account ({%cite genome-annotation-repeatmasker %}).

Reviewer 2: ```suggestion
Repeat-rich regions can interfere with genome annotation tools. In this step we find and soft-mask repetitive regions in the genome. The annotation tool can then take this information into account ([see RepeatMasker tutorial for more details]({% link topics/genome-annotation/tutorials/repeatmasker/tutorial.md %})).
```

We use **RepeatMasker**
{% cite RepeatMasker %}, a program that screens DNA sequences for interspersed repeats and low complexity DNA sequences.
This program has historically made use of [RepBase](https://www.girinst.org/repbase/update/index.html) ({%cite Kohany2006-ks %}), a service of the Genetic Information Research Institute, but this database in no longer open access. Instead, we will use [Dfam](https://www.dfam.org/home) ({%cite Storer2021 %}) an open collection of Transposable Element DNA sequence alignments,  HMMs derived from Repbase sequences and consensus sequences. For this reason, the annotation of repetitive sequences might be incomplete.

### FASTA pre-process with **Replace Text**

> RepeatMasker will only accept compact fasta headers. Before we can mask repetitive regions with RepeatMasker we must trim the NCBI long header (*">BK006939.2 TPA_inf: Saccharomyces cerevisiae S288C chromosome V, complete sequence"*) to leave only the accession number ("*>BK006939.2*") by using a regular expression.

Reviewer 2: ```suggestion
RepeatMasker will only accept compact fasta headers. Before we can mask repetitive regions with RepeatMasker we must trim the NCBI long header (`BK006939.2 TPA_inf: Saccharomyces cerevisiae S288C chromosome V, complete sequence`) to leave only the accession number (`>BK006939.2`) by using a regular expression.
```

>    - {% icon param-file %} *"Genomic DNA"*: `outfile` (output of **Replace Text** {% icon tool %})
>    - *"Repeat library source"*: `DFam (curated only, bundled with RepeatMasker)`
>        - *"Select species name from a list?"*: `No`
>            - *"Repeat source species"*: `"Saccharomyces cerevisiae"`
>
>    > ### {% icon comment %} Comment
>    >
>    > In principal, all unique clade names occurring in [NCBI taxonomy database](https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html) can be used for species. Capitalization is ignored, multiple words need to bound by apostrophes. Not all "common" English names occur in the taxonomy database. Using Latin names is always safest.
>    {: .comment}
>
{: .hands_on}

## Annotate with Funannotate
We will predict protein-coding genes from genomic sequences using [Funannotate](https://funannotate.readthedocs.io/) ({% cite Young2019 %}), which collects evidence from different ab-initio gene predictors as well as from RNA-seq or ESTs data. Funannotate has been developed for Fungi but it works with any Eukaryotic genome. The output of Funannotate is a list of ORFs and their translation in GenBank format.

> If you would like to learn about genome annotation in more depth, the GTN has a [section]({{ site.baseurl }}/topics/genome-annotation) dedicated to training on genome annotation, including a hands-on tutorial on [Funannotate]({{ site.baseurl }}/topics/genome-annotation/tutorials/funannotate/tutorial.html) ({%cite genome-annotation-funannotate %}).

Reviewer 2: ```suggestion
If you would like to learn about genome annotation in more depth, the GTN has a [section]({{ site.baseurl }}/topics/genome-annotation) dedicated to training on genome annotation, including a hands-on tutorial on [Funannotate]({% link topics/genome-annotation/tutorials/funannotate/tutorial.md %}).
```

same as the other time, no proper citation needed, fine to just link to it.

> **Even for a small dataset, Funannotate can take a very long time to run. You can skip this step and use the Genbank files downloaded from Zenodo for the following step. These were generated using Funannotate as described in the hands-on below.**

Reviewer 2: ```suggestion
> ### {% icon warning %} Slow Step Ahead!
> Even for a small dataset, Funannotate can take a very long time to run. You can skip this step and use the Genbank files downloaded from Zenodo for the following step. These were generated using Funannotate as described in the hands-on below.
{: .warning}
```

we can use a nice tip box here instead, and then it's maybe nicer for users to read, and more visibly called out.

>    - {% icon param-file %} *"Select lines from"*: `listproteinorthograbproteins` (output of **Proteinortho grab proteins** {% icon tool %})
>    - In *"Check"*:
>        - {% icon param-repeat %} *"Insert Check"*
>            - *"Find Regex"*: `(>[^_]+).+`
>            - *"Replacement"*: `\1`
>
{: .hands_on}

> ### {% icon hands_on %} Hands-on: Align orthologs with ClustalW
>
> 1. {% tool [ClustalW](toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1) %} with the following parameters:
>    - {% icon param-file %} *"FASTA file"*: `out_file1` (output of **Regex Find And Replace** {% icon tool %})
>    - *"Data type"*: `Protein sequences`
>    - *"Output alignment format"*: `FASTA format`
>    - *"Output complete alignment (or specify part to output)"*: `Complete alignment`
>
{: .hands_on}

## Re-arrange

To create the template, each step of the workflow had its own subsection.

***TODO***: *Re-arrange the generated subsections into sections or other subsections.
Consider merging some hands-on boxes to have a meaningful flow of the analyses*

# Conclusion

> {:.no_toc}

Reviewer 2: ```suggestion
```
